docs/proposal/20250224-rosa-networking.md

```diff
@@ -0,0 +1,221 @@
+     namespace: default
+   status:
+     resources:
+       - resource: NATGateway1
+         id:
+         status: CREATE_IN_PROGRESS
+         reason: Eventual consistency check initiated
+       - resource: VPC
+         id: vpc-055edf3ebf27f18d6 
+         status: CREATE_COMPLETE
+         reason:
+       - resource: SecurityGroup
+         id:
+         status: CREATE_IN_PROGRESS
+         reason: Resource creation Initiated
+   ```
+   and will be reflecting the the values coming from [AWS CloudFormation API](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation#Client.DescribeStackEvents) (`resource`, `id`, `status` and `reason`).
+
+   `status.conditions` of the `RosaNetwork` resource will be consistent with the CAPA conventions, example of a successful network stack creation:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-20T14:45:26Z"
+       status: "True"
+       type: RosaNetworkReady
+   ```
+   Example of failed network stack creation:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-18T13:25:16Z"
+       status: "False"
+       type: RosaNetworkReady
+       severity: Error
+       reason: ReconciliationFailed
+       message: Insufficient privileges for ...
+   ```
+   Failed deletion:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-18T13:25:16Z"
+       status: "False"
+       type: RosaNetworkReady
+       severity: Error
+       reason: DeletionFailed
+       message: ...
+   ```
+   
+1. Modifications in the ROSA control plane CRD & reconciler so that it would be possible to reference the `RosaNetwork` resource from control plane:
+   ```
+   kind: ROSAControlPlane
+     metadata:
+       name: hcp01-control-plane
+       namespace: default
+   spec:
+     rosaNetworkRef:
+       name: hcp01-rosa-network
+   ```
+   Should the ROSA control plane CR contain reference to ROSA network, the reconciler will read the AWS region, AZ and subnet ids parameters from the ROSA network CR. The ROSA control plane should also be validated through a webhook so that it does not contain both the reference to `RosaNetwork` and the subnet ids and / or availability zones.
+
+1. New tests.
+
+### User Stories
+
+1. As a CAPA user, I want to be able to provision the network infrastructure for ROSA HCP.
+2. As a CAPA user, I want to be able to use the provisioned network infrastructure in ROSA HCP control plane.
+3. As a CAPA user, I want to be able to delete the network infrastructure previously provisioned by CAPA.
+
+#### Functional Requirements
+
+1. Ability to create a new namespaced custom resource `RosaNetwork` with four attributes: name, AZ count, region and CIDR block for VPC.
```

Comment:
This should probably note the AZ list, too.

Reply:
@nrb yep, added to the proposal doc.